app/controllers/trainees_controller.rb

```diff
@@ -28,7 +28,7 @@ def create
     if trainee_params[:record_type] == "other"
       redirect_to trainees_not_supported_route_path
     else
-      authorize @trainee = Trainee.new(trainee_params.merge(provider_id: current_user.provider_id))
+      authorize @trainee = Trainee.new(trainee_params.merge(provider_id: current_user.provider_id, slug: SecureRandom.base58(24)))
```

Comment:
Not a big fan of ActiveRecord callbacks, but perhaps there's a good case to put the slug generation in an `before_create` callback? Not something the controller should know about. This will also get rid of `SecureRandom.base58(24)` calls in some view components and specs.

Reply:
I dislike callbacks (with a passion). The unintended side effects make testing difficult so usually try to avoid them.

I'd prefer we call regenerate_slug here on the trainee object before it's created so the controller remains unaware of what value to pass in for this field and just send a message to do the slug generation.

Reply:
I went with a callback 🤦 , the Trainee class is starting to get a bit beefy. The view component specs should ideally just work with a `double` or `OpenStruct`. `Trainee` is starting to become a little loaded. I've done this in https://github.com/DFE-Digital/register-trainee-teachers/pull/405/commits/9e311434810f371cafa169c0e7559182064629a3